internal/pkg/daemon/bpfrecorder/bpfrecorder.go

```diff
@@ -721,7 +721,9 @@ func (b *BpfRecorder) findContainerID(id string) error {
 
 			for p := range pods.Items {
 				pod := &pods.Items[p]
-				for c := range pod.Status.ContainerStatuses {
+				//nolint:gocritic // We explicitly do not want to append to the same slice
+				statuses := append(pod.Status.ContainerStatuses, pod.Status.InitContainerStatuses...)
+				for c := range statuses {
```

Comment:
FWIW, this was the isse, we generated longer statuses and iterated over them, but inside the for loop we would have still assigned `containerStatus := pod.Status.ContainerStatuses[c]`.

Reply:
The upcoming patch version changes to `containerStatus := statuses[c]`